cpmpy/expressions/globalconstraints.py

```diff
@@ -292,6 +304,18 @@ def deepcopy(self, memodict={}):
         copied_args = self._deepcopy_args(self.args)
         return Minimum(copied_args)
 
+    def decompose_comparison(self, cpm_op, cpm_rhs):
+        """
+        Decomposition if it's part of a comparison
+        """
+        from .python_builtins import any, all
+
+        arr = argval(self.args)
+        return [any(eval_comparison(cpm_op, j, cpm_rhs) & (all([j <= a for a in arr])) for j in arr)]
```

Comment:
- why use bit-wise `&` instead of `and`?
- this is a quadratic blow-up. We can avoid this by introducing an auxiliary symbol, say `_min` where `any(x>=_min for x in arr) and all(x<=_min for x in arr)` and then just assert `eval_comparison(cpm_op,x,cpm_rhs)`.

Reply:
I implemented this change, but for the auxiliary symbol we need to calculate bounds. So will only work properly after we added get_bounds to expressions (as I am working on in branch Bound_calculations)